astropy/io/fits/column.py

```diff
@@ -1389,16 +1387,13 @@ def __getattr__(self, name):
 
         Implements for example self.units, self.formats, etc.
         """
-
         cname = name[:-1]
         if cname in KEYWORD_ATTRIBUTES and name[-1] == 's':
-            attr = []
-            for col in self:
-                val = getattr(col, cname)
-                if val is not None:
-                    attr.append(val)
-                else:
-                    attr.append('')
+            attr = [getattr(col, cname) for col in self.columns]
+            if None in attr:
```

Comment:
Can't remember my specific example... Maybe my case was one of the ones explained by http://stackoverflow.com/questions/22108488/are-list-comprehensions-and-functional-functions-faster-than-for-loops